test/VoucherHub.test.ts

```diff
@@ -659,6 +664,86 @@ describe('VoucherHub', function () {
         });
     });
 
+    describe('Top up voucher', function () {
+        let [voucherOwner1, anyone]: SignerWithAddress[] = [];
+        let voucherHub: VoucherHub;
+        let voucherAddress: string;
+
+        beforeEach(async function () {
+            ({ voucherHub, voucherOwner1, anyone } = await loadFixture(deployFixture));
+            await voucherHubWithAssetEligibilityManagerSigner
+                .createVoucherType(description, duration)
+                .then((tx) => tx.wait());
+            voucherAddress = await voucherHubWithVoucherManagerSigner
+                .createVoucher(voucherOwner1, voucherType, voucherValue)
+                .then((tx) => tx.wait())
+                .then(() => voucherHub.getVoucher(voucherOwner1));
+        });
+
+        it('Should top up voucher', async function () {
+            const topUpValue = 123n; // arbitrary value
+            const voucherCreditBalanceBefore = await voucherHub.balanceOf(voucherAddress);
+            const voucherRlcBalanceBefore = await iexecPocoInstance.balanceOf(voucherAddress);
+
+            const tx = await voucherHubWithVoucherManagerSigner.topUpVoucher(
+                voucherAddress,
+                topUpValue,
+            );
+            const txReceipt = await tx.wait();
+            const expectedExpiration = await commonUtils.getExpectedExpiration(duration, txReceipt);
+            await expect(tx)
+                .to.emit(voucherHub, 'VoucherToppedUp')
+                .withArgs(voucherAddress, topUpValue, expectedExpiration);
+            const voucherCreditBalanceAfter = await voucherHub.balanceOf(voucherAddress);
+            const voucherRlcBalanceAfter = await iexecPocoInstance.balanceOf(voucherAddress);
+            expect(voucherCreditBalanceAfter)
+                .equal(voucherCreditBalanceBefore + topUpValue)
+                .equal(voucherRlcBalanceBefore + topUpValue)
+                .equal(voucherRlcBalanceAfter);
+            expect(
+                await Voucher__factory.connect(voucherAddress, anyone).getExpiration(),
+            ).to.be.equal(expectedExpiration);
```

Comment:
Also compare to "before" expiration.

Reply:
Please see [Compare to "before" expiration](https://github.com/iExecBlockchainComputing/iexec-voucher-contracts/pull/23/commits/a9f3362360a05b1d8bc6a6256280b97caa69ae28) :)